### DISCUSSION TITLE
Parent relative shader includes

### DISCUSSION THREAD

**Discussion by tomdrinKEENSWH** [amerkoleci/Vortice.Windows#199] [Q&A] [answered]
Hello,
we have previously used SharpDX where shader compiler's include handler had following signature:  
**Stream Open(IncludeType type, string fileName, Stream parentStream)**

This reference to parentStream allowed us to obtain parent filepath and therefore to implement relative includes.
For example:
- entry.hlsl with "#include ../file2.hlsli"
- ../file2.hlsli with "#include ../file3.hlsli" _//there the relative path to entry.hlsl is ../../file3.hlsl_

In Vortice the Include Handler api is simpler:
**Result LoadSource(string includeFileName, out IDxcBlob? includeSource)**

Would you consider adding the SharpDX style Include handlers?
Is there a clean way how to implement the described behaviour using current Include handlers?

**Comment by amerkoleci** [comment, marked as answer]
What you're talking is the legacy FXC include handler (ID3DInclude), the implementation in Vortice is based on IDXCIncludeHandler.
So the concept/usage are different and you should look at DirectXShaderCompiler usage.
